### PR TITLE
[google_maps_flutter] LatLngBorders

### PR DESCRIPTION
**Pull request by Soundbytes**
This is a thorough rewrite of the LatLngBounds class. It is based on js code that I shamelessly lifted from  the google maps javascript api (https://maps.googleapis.com/maps/api/js)
I needed a working extend method and this is the result. Hope you find it worthwhile.

### CODE REVIEW

**Comment by kf6gpe** [comment]
cc @amirh --- mind taking a look, please, as well?

@Soundbytes Thanks for the contribution! Do you have time to add a test for the new methods? If not, we'll get to it eventually.

**Comment by Soundbytes** [comment]
I have now added some tests. It appears that one fringe case even fails.
When uniting two bounds that together span around the globe union should return a full circle bounds which it doesn't .

I would expect this same issue to be present in the original js api since my code is based on that. I am afraid I won't have the time to check that out and file a bug report for the js api issue . 

I also have been assigned a new project and won't have much time to clean up the failing checks that have shown up with the introduction of the tests. (At least it will take some time).

**Comment by Soundbytes** [comment]
Looks like your mighty bots are happy now with my contribution. I am looking forward to seeing it included in the main branch. If there is anything else I can do please let me know.

BTW: I solved the one failing test by just removing it. A LatLngBounds that spans around the globe isn't really a thing at all. I don't think it's worth the effort since nobody will ever run into it.

**Comment by Soundbytes** [comment]
Congrats to you guys for finally putting the plugin up on pub. 
Are you considering to merge my branch any time soon? 
I am sure it will be a helpful addition since it implements the expected interface to the class.
(Plus I would love to revert to the official supported version of the ackage instead of using my branch.)
Thanks!

**Comment by amirh** [comment]
HI, I apologize for the delay, I'll try to get to this soon.
Would you mind adding Dartdocs for the public API? that will also help me getting started with the review.

**Comment by Soundbytes** [comment]
Sorry it took so long to fix the CLA issue and remove the falsely added commits. I am not really a git wizard.

**Comment by Soundbytes** [comment]
@amirh any news?

**Comment by NarenderKhod** [comment]
Hi,
I want to create a circle on google map but I didn't found any option on flutter google map.
So can you please let me know how can I achieve this one?

**Comment by NarenderKhod** [comment]
Hi,
I want to add a circle on google map using the google-map flutter package. I didn't found any option in this package. So can you please let me know how can I achieve this one.

**Comment by amirh** [comment]
cc @iskakaushik

**Comment by iskakaushik** [comment]
@NarenderKhod could you please file an issue for the circle. This PR is unrelated to it.

File: packages/google_maps_flutter/lib/src/location.dart [hunk 1]
```diff
@@ -60,47 +175,124 @@ class LatLng {
...
 class LatLngBounds {
-  /// Creates geographical bounding box with the specified corners.
-  ///
-  /// The latitude of the southwest corner cannot be larger than the
-  /// latitude of the northeast corner.
-  LatLngBounds({@required this.southwest, @required this.northeast})
-      : assert(southwest != null),
-        assert(northeast != null),
-        assert(southwest.latitude <= northeast.latitude);
+  /// Constructs a rectangle from the points at its
```
Comment (iskakaushik) [inline packages/google_maps_flutter/lib/src/location.dart]: can you leave the formatting as is? It renders in a better way and also follows: https://www.dartlang.org/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph

**Comment by iskakaushik** [comment]
Oh, please also rebase and fix the merge conflicts.

**Comment by iskakaushik** [comment]
@Soundbytes  do you plan on working on this PR?

**Comment by iskakaushik** [comment]
closing due to inactivity.

**Pull request closed without merging**